### PR TITLE
feat(hooks-store): update custom hook-dapps manifests periodically

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

In [this PR](https://github.com/cowprotocol/cowswap/pull/4910), I added a functionality to import custom hook-dapps using iframes.
This mechanism fetches manifest.json and save it content into localStorage.
As @yvesfracari noticed, the saved info never gets updated.
To fix that I added `IframeDappsManifestUpdater` which tries to update all custom hook-dapps, but not often than 6 hours.

# To Test

1. Add a custom hook-dapp (see the mentioned PR above)
2. Change something in its manifest (for example image)
3. Remove `HOOK_DAPPS_UPDATE_TIME` from localStorage
4. Update the page
- [ ] the image of the hook-dapp should be updated in CoW Swap

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @shoom3301 , works as described. 
But question: why should I remove an item from a local storage? Might it be updated on a page reload without this action? Again, sorry if it is too dumb question :)

**Comment by alfetopito** [comment]
> Hey @shoom3301 , works as described. But question: why should I remove an item from a local storage? Might it be updated on a page reload without this action? Again, sorry if it is too dumb question :)

Not dumb!

If you don't remove local storage, you'll have to wait for 6h. That's all.

**Review by elena-zh — APPROVED** [review]
@alfetopito , thank you for explanation! :)

File: apps/cowswap-frontend/src/modules/hooksStore/updaters/iframeDappsManifestUpdater.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  const { chainId } = useWalletInfo()
+
+  const [preHooks, postHooks] = useMemo(
+    () => [Object.values(hooksState.pre), Object.values(hooksState.post)],
+    [hooksState],
+  )
+
+  const fetchAndUpdateHookDapp = useCallback(
+    (url: string, isPreHook: boolean) => {
+      return fetch(`${url}/manifest.json`)
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/hooksStore/updaters/iframeDappsManifestUpdater.tsx:33]: It also will work, but will be a bit implicit comparing to the current implementation

**Pull request merged**